### PR TITLE
Fix different CLI commands throw an exception on help argument

### LINKED ISSUE #9005: Different CLI commands throws an Exception on help argument
Version: 7.6

**Describe the bug**
The help argument should only shows a short description/usage (supported arguments, ...)  and not throw a MissingOptionException

**To Reproduce**

Samples with Exception on help:
```
./dspace export --help
usage: export
 -d,--dest <arg>           destination where you want items to go
 -h,--help                 help
 -i,--id <arg>             ID or handle of thing to export
 -m,--migrate              export for migration (remove handle and
                           metadata that will be re-created in new system)
 -n,--number <arg>         sequence number to begin exporting items with
 -t,--type <arg>           type: COLLECTION or ITEM
 -x,--exclude-bitstreams   do not export bitstreams
 -z,--zip <arg>            export as zip file (specify filename e.g.
                           export.zip)
org.apache.commons.cli.MissingOptionException: Missing required options: t, i, d, n
        at org.apache.commons.cli.DefaultParser.checkRequiredOptions(DefaultParser.java:199)
        at org.apache.commons.cli.DefaultParser.parse(DefaultParser.java:130)
        at org.apache.commons.cli.DefaultParser.parse(DefaultParser.java:76)
        at org.apache.commons.cli.DefaultParser.parse(DefaultParser.java:60)
        at org.dspace.scripts.DSpaceRunnable.parse(DSpaceRunnable.java:85)
        at org.dspace.scripts.DSpaceRunnable.initialize(DSpaceRunnable.java:75)
        at org.dspace.app.launcher.ScriptLauncher.executeScript(ScriptLauncher.java:148)
        at org.dspace.app.launcher.ScriptLauncher.handleScript(ScriptLauncher.java:131)
        at org.dspace.app.launcher.ScriptLauncher.main(ScriptLauncher.java:98)
```

```
./dspace curate --help
...
org.apache.commons.cli.ParseException: Required parameter -e missing!
...
```

```
./dspace import --help
...
org.apache.commons.cli.MissingOptionException: Missing required options: m, e
...
```

Samples without Exception on help:
```
./dspace oai --help
OAI Manager Script
Syntax: oai <action> [parameters]
> Possible actions:
     import - To import DSpace items into OAI index and cache system
     clean-cache - Cleans the OAI cached responses
> Parameters:
     -o Optimize index after indexing (import only)
     -c Clear index (import only)
     -v Verbose output
     -h Shows this text
```

```
./dspace index-discovery --help
The script has started
usage: index-discovery
 -b,--build          (re)build index, wiping out current one if it exists
 -c,--clean          clean existing index removing any documents that no
                     longer exist in the db
 -d,--delete         delete all records from existing index
 -f,--force          if updating existing index, force each handle to be
                     reindexed even if uptodate
 -h,--help           print this help message
 -i,--index <arg>    add or update an Item, Collection or Community based
                     on its handle or uuid
 -r,--remove <arg>   remove an Item, Collection or Community from index
                     based on its handle
 -s,--spellchecker   Rebuild the spellchecker, can be combined with -b and
                     -f.
The script has completed

```

### PR DESCRIPTION
**Pull request by ChrisBethgster**
Fixes #9005

### CODE REVIEW

File: dspace-api/src/main/java/org/dspace/app/launcher/ScriptLauncher.java [hunk 1]
```diff
@@ -145,8 +146,13 @@ public static int handleScript(String[] args, Document commandConfigs,
     private static int executeScript(String[] args, DSpaceRunnableHandler dSpaceRunnableHandler,
                                      DSpaceRunnable script) {
         try {
-            script.initialize(args, dSpaceRunnableHandler, null);
-            script.run();
+            StepResult result = script.initialize(args, dSpaceRunnableHandler, null);
+            if (StepResult.Continue.equals(result)) {
+                // only run the script, if the normal initialize is successful
+                script.run();
+            } else {
```
Comment (hardyoyo) [inline dspace-api/src/main/java/org/dspace/app/launcher/ScriptLauncher.java:153]: is this else really necessary?

**Comment by hardyoyo** [comment]
While this PR seems correct, in testing it, I ran the following command after starting up DSpace:

```
bin/dspace curate --help
usage: curate
 -e,--eperson <arg>     email address of curating eperson
 -h,--help              help
 -i,--id <arg>          Id (handle) of object to perform task on, or 'all'
                        to perform on whole repository
 -p,--parameter <arg>   a task parameter 'NAME=VALUE'
 -q,--queue <arg>       name of task queue to process
 -r,--reporter <arg>    relative or absolute path to the desired report
                        file. Use '-' to report to console. If absent, no
                        reporting
 -s,--scope <arg>       transaction scope to impose: use 'object',
                        'curation', or 'open'. If absent, 'open' applies
 -t,--task <arg>        curation task name; options: [noop,
                        profileformats, requiredmetadata, checklinks,
                        registerdoi]
 -T,--taskfile <arg>    file containing curation task names
 -v,--verbose           report activity to stdout

whole repo: CurationCli -t estimate -i all
single item: CurationCli -t generate -i itemId
task queue: CurationCli -q monthly
org.apache.commons.cli.ParseException: Required parameter -e missing!
	at org.dspace.curate.CurationCli.assignCurrentUserInContext(CurationCli.java:45)
	at org.dspace.curate.Curation.setup(Curation.java:230)
	at org.dspace.scripts.DSpaceRunnable.parse(DSpaceRunnable.java:121)
	at org.dspace.scripts.DSpaceRunnable.initialize(DSpaceRunnable.java:94)
	at org.dspace.app.launcher.ScriptLauncher.executeScript(ScriptLauncher.java:149)
	at org.dspace.app.launcher.ScriptLauncher.handleScript(ScriptLauncher.java:132)
	at org.dspace.app.launcher.ScriptLauncher.main(ScriptLauncher.java:99)
```

So... it *does* generate the help text now, but it still throws an error. I will see if I can suggest a change, but I thought I'd at least report this finding.

File: dspace-api/src/main/java/org/dspace/app/launcher/ScriptLauncher.java (hunk 1 above)
Comment (ChrisBethgster) [reply dspace-api/src/main/java/org/dspace/app/launcher/ScriptLauncher.java]: The else was only for documentation...

Else is removed the documentation refactored in 064e2caa37dfa283c3c08dee0e7321e36073bfa2

**Comment by hardyoyo** [comment]
Can confirm, the revised code works perfectly. 👍

**Review by hardyoyo — APPROVED** [review]
Confirmed that this code works correctly, and addresses #9005 👍

**Review by tdonohue — APPROVED** [review]
👍 Code looks good to me too. Thanks @ChrisBethgster for your contribution!

**Pull request merged**